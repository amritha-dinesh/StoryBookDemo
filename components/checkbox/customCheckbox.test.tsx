import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import CustomCheckBox from "./customCheckbox";

describe("CustomCheckbox", () => {
  it("should match the snapshot", () => {
    const { toJSON } = render(<CustomCheckBox />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("renders correctly", () => {
    const { getByTestId } = render(<CustomCheckBox />);
    const customCheckboxComponent = getByTestId("custom-checkbox");
    expect(customCheckboxComponent).toBeDefined();
  });

  it("should toggle the checkbox when clicked", () => {
    const onChangeMock = jest.fn();
    const { getByTestId } = render(
      <CustomCheckBox value={false} onValueChanged={onChangeMock} />
    );
    const checkbox = getByTestId("custom-checkbox");
    fireEvent.press(checkbox);
    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith(true);
  });

  it("should not toggle the checkbox when clicked", () => {
    const onChangeMock = jest.fn();
    const { getByTestId } = render(
      <CustomCheckBox
        value={false}
        disabled={true}
        onValueChanged={onChangeMock}
      />
    );
    const checkbox = getByTestId("custom-checkbox");
    fireEvent.press(checkbox);
    expect(onChangeMock).toHaveBeenCalledTimes(0);
  });
});
