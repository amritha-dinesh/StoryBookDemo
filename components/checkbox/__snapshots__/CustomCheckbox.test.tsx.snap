// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCheckbox should match the snapshot 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "alignItems": "center",
      },
      {
        "flexDirection": "row",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="custom-checkbox"
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/un_checked.png",
        }
      }
      style={
        {
          "height": 23,
          "width": 23,
        }
      }
    />
  </View>
  <Text
    style={
      {
        "color": "black",
        "fontFamily": undefined,
        "fontSize": undefined,
        "marginHorizontal": 10,
      }
    }
  >
    label
  </Text>
</RCTSafeAreaView>
`;
